src/System.Data.SqlClient/tests/ManualTests/SQL/ConnectionPoolTest/PoolBlockPeriodTest.netcoreapp.cs

```diff
@@ -0,0 +1,205 @@
+
+        [Theory]
+        [InlineData("Test policy with Auto (lowercase)", new object[] { "auto" })]
+        [InlineData("Test policy with Auto (miXedcase)", new object[] { "auTo" })]
+        [InlineData("Test policy with Auto (Pascalcase)", new object[] { "Auto" })]
+        public void TestSetPolicyWithAutoVariations(string description, object[] Params)
+        {
+            SqlConnection.ClearAllPools();
+            string policyString = Params[0] as string;
+
+            string connString = CreateConnectionString(_sampleAzureEndpoint, null) + $";{_policyKeyword}={policyString}";
+            PoolBlockingPeriodAzureTest(connString, PoolBlockingPeriod.Auto);
+        }
+
+        [Theory]
+        [InlineData("Test policy with Always (lowercase)", new object[] { "alwaysblock" })]
+        [InlineData("Test policy with Always (miXedcase)", new object[] { "aLwAysBlock" })]
+        [InlineData("Test policy with Always (Pascalcase)", new object[] { "AlwaysBlock" })]
+        public void TestSetPolicyWithAlwaysVariations(string description, object[] Params)
+        {
+            SqlConnection.ClearAllPools();
+            string policyString = Params[0] as string;
+
+            string connString = CreateConnectionString(_sampleAzureEndpoint, null) + $";{_policyKeyword}={policyString}";
+            PoolBlockingPeriodAzureTest(connString, PoolBlockingPeriod.AlwaysBlock);
+        }
+
+        [Theory]
+        [InlineData("Test policy with Never (lowercase)", new object[] { "neverblock" })]
+        [InlineData("Test policy with Never (miXedcase)", new object[] { "neVeRblock" })]
+        [InlineData("Test policy with Never (Pascalcase)", new object[] { "NeverBlock" })]
+        public void TestSetPolicyWithNeverVariations(string description, object[] Params)
+        {
+            SqlConnection.ClearAllPools();
+            string policyString = Params[0] as string;
+
+            string connString = CreateConnectionString(_sampleNonAzureEndpoint, null) + $";{_policyKeyword}={policyString}";
+            PoolBlockingPeriodNonAzureTest(connString, PoolBlockingPeriod.NeverBlock);
+        }
+
+        public void PoolBlockingPeriodNonAzureTest(string connStr, PoolBlockingPeriod? policy)
+        {
+            SqlConnection.ClearAllPools();
+            Guid previousConnectionId = Guid.Empty;
+            int count = 0;
+
+            while (count < 2)
+            {
+                using (SqlConnection sqlConnection = new SqlConnection(connStr))
+                {
+                    try
+                    {
+                        sqlConnection.Open();
+                        throw new Exception("Connection Open must expect an exception!");
+                    }
+                    catch (SqlException e)
+                    {
+                        // if it is the first time the exception is happening (previousConnectionId == Guid.Empty) skip the check
+                        if (previousConnectionId != Guid.Empty)
```

Comment:
Here's my understanding: 
When the SqlConnection is unable to establish a TCP connection, it sets an Empty Guid for the SqlConnection Connection Id. In your test you are setting `previousConnectionId = e.ClientConnectionId` which is effectively Guid.Empty.  
Is your test entering this if condition?